src/AbstractNode.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the KongAdminApi package.
+ *
+ * (c) Unikorp <https://github.com/unikorp>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Unikorp\KongAdminApi;
+
+/**
+ * abstract node
+ *
+ * @author VEBER Arnaud <https://github.com/VEBERArnaud>
+ */
+abstract class AbstractNode implements NodeInterface
+{
+    const JSON_CONTENT_TYPE = [
+        'Content-Type' => 'application/json',
+    ];
+
+    /**
+     * client
+     * @var \Unikorp\KongAdminApi\Client $client
+     */
+    private $client = null;
+
+    /**
+     * Constructor
+     *
+     * @param \Unikorp\KongAdminApi\Client $client
+     */
+    public function __construct(Client $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * get
+     *
+     * @param string $endpoint
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    protected function get($endpoint)
+    {
+        return $this->client->getHttpClient()->get($endpoint);
+    }
+
+    /**
+     * post
+     *
+     * @param string $endpoint
+     * @param DocumentInterface $document
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    protected function post($endpoint, DocumentInterface $document = null)
+    {
+        $body = (!is_null($document)) ? $document->toJson() : '[]';
+
+        return $this->client->getHttpClient()->post($endpoint, self::JSON_CONTENT_TYPE, $body);
+    }
+
+    /**
+     * put
+     *
+     * @param string $endpoint
+     * @param DocumentInterface $document
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    protected function put($endpoint, DocumentInterface $document = null)
+    {
+        $body = (!is_null($document)) ? $document->toJson() : '[]';
+
+        return $this->client->getHttpClient()->put($endpoint, self::JSON_CONTENT_TYPE, $body);
+    }
+
+    /**
+     * patch
+     *
+     * @param string $endpoint
+     * @param DocumentInterface $document
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    protected function patch($endpoint, DocumentInterface $document = null)
+    {
+        $body = (!is_null($document)) ? $document->toJson() : '[]';
+
+        return $this->client->getHttpClient()->patch($endpoint, self::JSON_CONTENT_TYPE, $body);
+    }
+
+    /**
+     * delete
+     *
+     * @param string $endpoint
+     * @param DocumentInterface $document
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    protected function delete($endpoint, DocumentInterface $document = null)
+    {
+        $body = (!is_null($document)) ? $document->toJson() : '[]';
```

Comment:
I suggest to remove parenthesis which are not needed and use a more simple syntax:

```
$body = $document ? $document->toJson() : '[]';
```